tests/h/views/api/helpers/links_test.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import pytest
+
+from h.views.api.helpers import links
+
+
+class TestRegisterLink(object):
+    def test_it_creates_attrs_on_registry_if_not_present(
+        self, service_link, versions, pyramid_config
+    ):
+        links.register_link(service_link(), versions, pyramid_config.registry)
+
+        assert hasattr(pyramid_config.registry, "api_links")
+        assert "v1" in pyramid_config.registry.api_links
+        assert "v2" in pyramid_config.registry.api_links
+
+    def test_it_registers_link_for_every_version(
+        self, service_link, versions, pyramid_config
+    ):
+        link = service_link()
+        links.register_link(link, versions, pyramid_config.registry)
+        assert link in pyramid_config.registry.api_links["v1"]
+        assert link in pyramid_config.registry.api_links["v2"]
+
+    def test_it_does_not_register_link_for_unsupported_versions(
+        self, service_link, versions, pyramid_config
+    ):
+        first_service = service_link()
+        second_service = service_link("doodad")
+
+        links.register_link(first_service, versions, pyramid_config.registry)
+        links.register_link(second_service, ["v1"], pyramid_config.registry)
+
+        assert first_service in pyramid_config.registry.api_links["v2"]
+        assert second_service not in pyramid_config.registry.api_links["v2"]
+
+    def test_it_registers_primary_method_on_link(
+        self, service_link, versions, pyramid_config
+    ):
+        link = service_link()
+
+        links.register_link(link, versions, pyramid_config.registry)
+
+        assert pyramid_config.registry.api_links["v1"][0]["method"] == "POST"
+
+    def test_it_registers_first_primary_method_if_multiple(
+        self, service_link, versions, pyramid_config
+    ):
+        multi_link = service_link()
+        multi_link["method"] = ("GET", "POST")
+
+        links.register_link(multi_link, versions, pyramid_config.registry)
+
+        assert pyramid_config.registry.api_links["v1"][0]["method"] == "GET"
+
+
+@pytest.fixture
+def service_link():
+    def make_service_link(name=None):
+        return {
+            "name": name or "api.example_service",
+            "description": "Create a new Foo",
+            "route_name": "api.example_service",
+            "method": "POST",
+        }
+
+    return make_service_link
```

Comment:
This looks like it actually a test helper function not a fixture:

```python
@pytest.fixture
def example_service_link():
    return _service_link("api.example.service")    

def _service_link(name):
    return {
        "name": name,
        "description": "Create a new Foo",
        "route_name": "api.example_service",
        "method": "POST",
    }
```

Then tests can take `example_service_link` as an argument or can just call `_service_link("foo")` directly (without using it as a fixture)

Reply:
Ah. Interesting. I've seen this done as fixtures elsewhere. I can convert it to a helper function.